### ISSUE TITLE
MetaAlmanac not loading planetary data for 301 (Earth's Moon)

### ISSUE THREAD

**Issue by translunar** [nyx-space/anise#339]
# Bug report

## Describe the bug
I cannot access lunar planetary constants or frame data through anise.

## To Reproduce
```
import pathlib
import anise

def load_kernels() -> anise.MetaAlmanac:
    my_constants_filename = (
        'my_constants.tpc')
    earth_assoc_itrf93_fk_filename = (
        'earth_assoc_itrf93.tf')
    earth_assoc_itrf93_epa_filename = (
        'earth_assoc_itrf93.epa')

    if not pathlib.Path(my_constants_filename).is_file():
        # Make sure the constants file exists (create if not)
        anise.utils.convert_tpc(
                'pck00010.tpc',
                'gm_de440.tpc',
                my_constants_filename)

    if not pathlib.Path(earth_assoc_itrf93_epa_filename).is_file():
        anise.utils.convert_fk(
            earth_assoc_itrf93_fk_filename,
            earth_assoc_itrf93_epa_filename)

    earth_latest_high_prec = anise.MetaFile(
        'earth_latest_high_prec.bpc',
    )
    jpl_moon_rotation = anise.MetaFile(
        'moon_pa_de440_200625.bpc',
    )
    de440 = anise.MetaFile(
        'de440.bsp'
    )
    moon_frames = anise.MetaFile(
        'moon_fk.epa'
    )
    itrf93 = anise.MetaFile(earth_assoc_itrf93_epa_filename)
    my_constants = anise.MetaFile(
        'my_constants.tpc'
    )
    new_meta = anise.MetaAlmanac()
    new_meta.files = [
        earth_latest_high_prec,
        jpl_moon_rotation,
        de440,
        itrf93,
        moon_frames,
        my_constants
    ]
    return almanac
```
and then `almanac.describe(spk=True, bpc=True, planetary=True)` does not list 301. I believe this should be in `de440.bsp`, but maybe I'm missing something. I spent a bunch of time staring at the documentation for MetaAlmanac and Almanac and couldn't figure this out.

## Expected behavior

I would expect loading the planetary constants and BPC files should yield planetary data for 301. I should also be able to get the MOON_J2000 frame, but I get the following error with `almanac.frame_info(Frames.MOON_J2000)`:

```
Exception: when fetching frame by its UID via ephemeris_id, when fetching by ID, ID 301 not in look up table
```

## Platform
Linux, Anise 0.4.4

**Comment by translunar** [comment]
Ah! It seems as if the real issue is that I wasn't using `almanac.frame_info` correctly.

**Comment by translunar** [comment]
I found a workaround, which is to load a pck11.pca, but this was hard to do because the tutorial link is still to the v0.3 version of that file, and so I had to do a little URL hacking to figure out what I needed to wget.

**Comment by ChristopherRabotin** [comment]
Hi Juno,

Thanks for the bug report. The tutorials need some updating to the paths, and I suspect the readme might still point to version 3 of the files: I'll fix this in #341.

Would it be possible for you to share the files used for creating this planetary constants input file? I'd like to reproduce the issue: from your code, it seems like everything should have worked.

Thanks

**Comment by translunar** [comment]
They are as listed: `pck00010.tpc` and `gm_de440.tpc`. These should be available from NAIF. Are you asking for some other file?

**Comment by ChristopherRabotin** [comment]
Yes: would you have a copy of the `my_constants.tpc` created by the call to `convert_tpc`? Both [`pck08.pca`](https://github.com/nyx-space/anise/blob/fc14d1be46fdbcb05ab87f0713f384b63279a388/data/pck08.pca) and [`pck11.pca`](https://github.com/nyx-space/anise/blob/fc14d1be46fdbcb05ab87f0713f384b63279a388/data/pck11.pca) are built using that same call in the tests, and they both include body 301.

PCK08:
![Image](https://github.com/user-attachments/assets/1d96d3bd-3520-4453-a07e-247ccc6e71d4)

PCK11:
![Image](https://github.com/user-attachments/assets/38ef5173-24d3-40f9-b24c-8c93ce55ca33)

**Comment by translunar** [comment]
Yeah. Oddly SPICE isn't loading 301 or 399 from pck 10 either, though I see the variables in the file. Here is the output file: 
[constants.tpc.txt](https://github.com/user-attachments/files/17421655/constants.tpc.txt)

And here is the input: [pck00010.tpc.txt](https://github.com/user-attachments/files/17421658/pck00010.tpc.txt)

**Comment by ChristopherRabotin** [comment]
Hey Juno,

Has the pck 11 or 08 versions been working for you? I can investigate this issue further at a later date. I'm also a bit surprised that the data present neither in SPICE nor ANISE since it sure looks present in the original tpc file.

**Comment by gregjesl** [comment]
```rust
let ctx = Almanac::new("./almanac/de440s.bsp").expect("Could not load bsp");
ctx.load("./almanac/moon_pa_de440_200625.bpc").expect("Could not load bpc");
println!("Planetary data length: {}", ctx.planetary_data.len());
```
reports
```bash
Planetary data length: 0
```

It looks like `Almanac::new` and `Almanac::load` are returning `Ok` despite not actually loading any data. 

For reference, my download script:
```bash
mkdir -p almanac
cd almanac
curl http://public-data.nyxspace.com/anise/de440s.bsp -O
curl http://public-data.nyxspace.com/anise/v0.4/pck11.pca -O
curl http://public-data.nyxspace.com/anise/v0.4/moon_fk.epa -O
curl http://public-data.nyxspace.com/anise/moon_pa_de440_200625.bpc -O
curl https://naif.jpl.nasa.gov/pub/naif/generic_kernels/pck/earth_latest_high_prec.bpc -O
```

**Comment by ChristopherRabotin** [comment]
Hi Greg,

The planetary data in the almanac refers only to the data you'd typically find the SPICE PCK files, including gravity parameters and shape of the ellipsoid. The `ctx.planetary_data.len()` returns the length of that dataset. So you need to also load the `pck11.pca` file.

```rust
let ctx = Almanac::new("./almanac/de440s.bsp").expect("Could not load bsp");
let loaded_ctx = ctx.load("./almanac/moon_pa_de440_200625.bpc").expect("Could not load bpc").load("./almanac/pck11.pca").expect("failure pck");

println!("Planetary data length: {}", loaded_ctx.planetary_data.len());
```

That will print `Planetary data length: 54`.

The BPC file only contains the data to rotate from one frame to another. In the case of the Moon PA file, it allows the rotation between the 31008 frame ID and the inertial frame:

![Image](https://github.com/user-attachments/assets/78e0574a-882f-4b2d-9ceb-6926ea8a468f)

(This is a screenshot from the ANISE GUI.)

**Comment by gregjesl** [comment]
> let ctx = Almanac::new("./almanac/de440s.bsp").expect("Could not load bsp");
> let loaded_ctx = ctx.load("./almanac/moon_pa_de440_200625.bpc").expect("Could not load bpc").load("./almanac/pck11.pca").expect("failure pck");

That worked for me, thanks! In retrospect, I think I had two issues going on: 
- At first, I was loading all off the files but didn't realize that `load` returns the loaded instance (I was trying to use `ctx` instead of `loaded_ctx` later in the routine)
- When I attempted to recreate it and stored `loaded_ctx`, I didn't load all the files.

**Comment by translunar** [comment]
I'm still running into problems with this.

```
(Pdb) ctx.load('spice/kernels/pck11.pca')
*** Exception: with astranis/gnc/spice/kernels/pck11.pca: Provided file cannot be inspected loaded directly in ANISE and may need a conversion first
```
This is the PCK file I got directly from Anise.

**Comment by translunar** [comment]
Okay, I think I narrowed down the problem. When I comment out the `convert_tpc` code, and just load my constants directly from `pck11.pca`, I can get EARTH_J2000 and EARTH_ITRF93 info.

**Comment by translunar** [comment]
Hmm. I was going to write you a unit test, but I can't actually make the existing unit test (test_almanac.py) work. Is there a reason you're not using Python's `unittest.TestCase` and pytest here?

**Comment by ChristopherRabotin** [comment]
The only reason I'm not using `unittest.TestCase` is because any function that starts with `test_` can be found by pytest without more of a set up, and the Pythontesting focuses on checking that functions can be called correctly; the Rust tests are the actual unit and integration tests.

That said, if you'd like to code up a reproducible issue using the unittest framework, I'll gladly take it and keep in the python test suite for regression testing.

**Comment by translunar** [comment]
Oh, okay, I didn't realize that worked like that. Well, I'm getting an error in `test_state_transformation` about missing `de440s.bsp`. When I put that file in the `data` directory, it works. I'll see if I can throw together a unit test for this.

**Comment by translunar** [comment]
Okay, just pushed a unit test. Looks like it's gm_de440.tpc that is the source of the problem. I tried it with the earlier version you have in the repo and it worked fine.
